### PR TITLE
[feature] #3274: Add to `kagami` a subcommand that generates examples

### LINKED ISSUE #3274: Generate a JSON file with the last structures of all ISI
### Feature request

Generate a JSON file containing the last examples structures of all existing ISI instructions from the code located at core/src/smartcontracts/isi/ for testing purposes, similar to the genesis.json? 

### Motivation

I require this to construct user's instructions that have not yet been added to the client_cli.

### Who can help?

@astrokov7

### PR DESCRIPTION
**Pull request by pesterev**
## Description

Add a new subcommand to `kagami` that generates an example by the chosen use case.

Another identical PR but aimed at [the LTS branch](https://github.com/hyperledger/iroha/pull/3398).

### Linked issue

Closes #3274  

### Benefits

This feature will be helpful for our QA team, at least, and will reduce the complexity of using the `iroha_client_cli json` command.  

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [x] Update kagami's README.md
- [x] All applicable CI checks pass (or I promised to make them pass later)
- [x] Test with `client_cli`
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

**Review by appetrosyan — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

File: tools/kagami/src/main.rs
Comment (QuentinI) [inline tools/kagami/src/main.rs:826]: What do you mean by "default"?

**Review by QuentinI — APPROVED** [review]

**Review by QuentinI — APPROVED** [review]

File: tools/kagami/src/main.rs
Comment (0x009922) [inline tools/kagami/src/main.rs:827]: It is an argument/option rather than a flag. Booleans are called flags.

```suggestion
            /// If this argument isn't provided, will generate a new key pair.
```

It is also relevant for further doc comments.

File: tools/kagami/README.md
Comment (0x009922) [inline tools/kagami/README.md:207]: There was https://github.com/hyperledger/iroha/pull/3316, which transformed all keys in the repo to a new format: lowercase multihash plus uppercase hex. Shouldn't it be same here too?

File: tools/kagami/src/main.rs
Comment (0x009922) [inline tools/kagami/src/main.rs:823]: Shouldn't users know what will happen if this option is omitted, like with `--public-key`?

File: tools/kagami/src/main.rs
Comment (0x009922) [inline tools/kagami/src/main.rs:791]:
```suggestion
        use_case: UseCase,
```

File: tools/kagami/src/main.rs
Comment (0x009922) [inline tools/kagami/src/main.rs:60]:
```suggestion
    /// Generate an example instruction based on the chosen use case
```

File: tools/kagami/src/main.rs
Comment (0x009922) [inline tools/kagami/src/main.rs:848]: Why other instructions support omitting parameters while this one doesn't?

File: tools/kagami/README.md
Comment (pesterev) [reply tools/kagami/README.md:207]: should be I guess

File: tools/kagami/src/main.rs
Comment (pesterev) [reply tools/kagami/src/main.rs:848]: In my opinion, we must know precisely what we will burn here. For example, when creating an account, we sometimes want to generate a new key pair, and we don't care what `account_id` uses for that.

**Comment by pesterev** [comment]
Let's not merge it while we consider [my suggestion](https://github.com/hyperledger/iroha/issues/3274#issuecomment-1525111817)

File: tools/kagami/src/main.rs
Comment (ilchu) [reply tools/kagami/src/main.rs:60]: Would appreciate the same wording in `README.md`, should help with clarity.

File: tools/kagami/src/main.rs
Comment (ilchu) [reply tools/kagami/src/main.rs:823]: The same would be helpful for the mint variant as well.

**Review by ilchu — COMMENTED** [review]
Can approve after minor fixes. Regarding the issue's [comment](https://github.com/hyperledger/iroha/issues/3274#issuecomment-1525111817), I believe this can be solved in a separate issue, as this one would need to be merged anyways just like it was in the LTS.

**Pull request closed without merging**
